transport.go

```diff
@@ -106,12 +111,23 @@ func NewTransport(key ic.PrivKey, psk pnet.PSK) (tpt.Transport, error) {
 	if err != nil {
 		return nil, err
 	}
+	config := quicConfig.Clone()
+	keyBytes, err := key.Raw()
+	if err != nil {
+		return nil, err
+	}
+	keyReader := hkdf.Expand(sha256.New, keyBytes, []byte("libp2p quic stateless reset key"))
```

Comment:
Let's define the info as a constant.

Reply:
Also, I don't think we can do this as the key may not be _uniformly_ random. I think we need the "extract" step (or need to call `New`)

I wonder if we should just _define_ some form of "DeriveSecretKey(info)` function on our secret key interface. That way, we have a standard way to do this.

Reply:
Sounds good. Let me prepare a PR.

Reply:
@Stebalien Can we move forward with this PR without resolving https://github.com/libp2p/go-libp2p-core/pull/131 first? For the stateless reset token, the only thing that matter is that the peer has **any** was of deriving a key. No interoperability required.

Reply:
You're right.

Reply:
Note: if we do that, you'll need to update this patch to call `hkdf.New` instead of `hkdf.Expand`.

Reply:
@Stebalien Done.